### PR TITLE
SAR similarity computations

### PR DESCRIPTION
**Pull request by anargyri**
### Description
Rewrite of the similarity computations so that they are done in-place at the sparse matrices and there are no divisions by zero.

### Related Issues
#160 

### Checklist:

- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by yueguoguo** [comment]
@anargyri there are some conflicts.

**Comment by miguelgfierro** [comment]
it looks you have to update staging, merge staging to your branch and then commit again

File: reco_utils/common/python_utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import numpy as np
```
[the remark this answers is not in the document]
Comment (anargyri) [reply reco_utils/common/python_utils.py]: Well, I am following the naming convention other people use in this directory. I do not want to change this convention here, so if you want to change them you can do it in a new PR.

[the remark this answers is not in the document]
Comment (anargyri) [reply reco_utils/common/python_utils.py]: Good point.

File: reco_utils/common/python_utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from scipy.sparse import coo_matrix, csc_matrix
+
+
+def jaccard(cooccurrence):
+    """Helper method to calculate the Jaccard similarity of a matrix of cooccurrences
```
[the remark this answers is not in the document]
Comment (anargyri) [reply reco_utils/common/python_utils.py]: I prefer to have the precise definition in the docstring and include more high level, subjective statements in the deep dive notebook or in a markdown document.

[the remark this answers is not in the document]
Comment (miguelgfierro) [reply reco_utils/common/python_utils.py]: make sense

File: reco_utils/common/python_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (miguelgfierro) [reply reco_utils/common/python_utils.py]: ok fair enough

**Pull request merged**